### DISCUSSION TITLE
Syntax highligthing on github and gitlab

### DISCUSSION THREAD

**Discussion by uramer** [teal-language/tl#601] [General]
Currently on github Teal seems to be highlighted in the same way as plain Lua. github uses Linguist https://github.com/github/linguist for highlighting, so support for Teal would need to be added there.

On gitlab it is not highlighted at all. Gitlab uses rouge https://github.com/rouge-ruby/rouge/wiki/Adding-a-new-lexer for highlighting

**Comment by alerque** [comment]
If there are any VSCode compatible syntax highlighters for Teal it isn't that hard to get linguist added. The process is a bit confusing but at the end of the day it can mostly reuse rulesets from existing lexers. I successfully added Fluent not to long ago, working forward from existing systax highlighter projects.

No experience wit rouge.

Either way this is not something this repo or even project needs to support, anybody can do the work to get this setup.

**Reply by virusdefender** [reply]
https://github.com/teal-language/vscode-teal/tree/master

**Reply by alerque** [reply]
That should do the trick. All that remains is to jump through all the hoops to get that processed as a module in the linguist build process.

**Comment by Aire-One** [comment]
I'd like to also mention https://github.com/alecthomas/chroma. This is what Gitea uses since https://github.com/go-gitea/gitea/pull/12047

**Reply by catwell** [reply]
For reference [Tangled](https://tangled.org) also uses Chroma through [goldmark-highlighting](https://github.com/yuin/goldmark-highlighting).

**Comment by jdnewman85** [comment]
Are there 200 unique projects using teal yet? [That's the first requirement listed for getting it on github](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-an-extension-to-a-language):
`We try only to add new extensions once they have some usage on GitHub. In most cases we prefer that each new file extension be in use in at least 200 unique :user/:repo repositories before supporting them in Linguist`

I haven't been able to craft a good search for this yet. Searching for `tlconfig.lua` makes it feel like we don't have _that_ many projects out there? Curious if someone can create a good query for this.

Additionally since the `tl` extension is used by other languages, this adds additional burden/steps in having to ensure their system can distinguish between the languages.

However, as mentioned above, we should be able to use the vscode-teal repo to accomplish this.

Certainly wouldn't hurt to get the PR made, and everything in place. Then it can wait for `popularity`

**Reply by hishamhm** [reply]
@jdnewman85 Thanks for the info!

I made a quick Github search and indeed we're still far from 200 projects. I think one good way to track these (and avoid false positives) is to start putting both the "lua" and "teal" tags in our repos.

**Reply by mtdowling** [reply]
> I made a quick Github search

What search did you use?

**Comment by Hedwig7s** [comment]
Got curious and had a look and there is a huge issue with this
Even when 200 projects is reached the file extension conflicts with another language ("type language")

**Reply by hishamhm** [reply]
Fortunately, this is not a huge issue, because the library can detect which one it is based on the presence of keywords. There are other languages that share file extensions.

**Reply by Hedwig7s** [reply]
Oh that's good
Would be sweet if it did get added (not sure if it meets the criteria but I can find lots of tlconfig mentions so it might)

**Comment by mtdowling** [comment]
Here's to hoping this one is merged: https://github.com/github-linguist/linguist/pull/7601

**Reply by mtdowling** [reply]
The PR was merged! Teal will have GitHub syntax highlighting whenever the next Linguist release is deployed on GitHub (typically 3-4 months, last release was 3 weeks ago).

**Reply by hishamhm** [reply]
@mtdowling This is excellent news! Thank you for driving this!!

**Reply by Hedwig7s** [reply]
That's one down, but the other 2 (gitlab and gitea) will be far more difficult as they can't reuse VSCode style highlighting

**Reply by Hedwig7s** [reply]
Looked into GitLab (Rouge) and it is probably going to be a bit of a mess because of how the lexer is set up
Most of the lexers on it are also rushed
